runtimebp/cpu.go

```diff
@@ -127,11 +129,84 @@ func defaultMaxProcsFormula(n float64) int {
 	return int(math.Ceil(n))
 }
 
-// GOMAXPROCS sets runtime.GOMAXPROCS with the default formula,
+func scaledMaxProcsFormula(n float64) int {
+	i := fetchMaxProcsMult()
+	return int(math.Ceil(n) * float64(i))
+}
+
+func fetchMaxProcsMult() int {
+	// Allow using a multiplier for number of processes relative to limit.
+	// Not catching the ok here because this function will never be called unless
+	// the environment variable is set
+	v, _ := os.LookupEnv("GOMAXPROCSMULT")
+	i, err := strconv.Atoi(v)
+	if err != nil {
+		fmt.Fprintf(
+			os.Stderr,
+			"runtimebp.fetchMaxProcsMult: GOMAXPROCSMULT is set to %q, which is invalid, falling back to 2 to ensure parallelism: %v",
+			v,
+			err,
+		)
+		i = 2
+	}
+	if i <= 0 {
+		i = 1
+	}
+	return i
+}
+
+// fetchCpuRequest checks for the magic CPUREQUEST variable set as a fallback
+// by Infrared.  If this is set, we use it and the multiplier to set our
+// GOMAXPROCS as a fallback.
+func fetchCPURequest() (int, error) {
+	if v, ok := os.LookupEnv("CPUREQUEST"); ok {
+		req, err := strconv.Atoi(v)
+		if err != nil {
+			// This case will probably never be hit, but I'm keeping it just in case.
+			// In k8s 1.21 and newer, when you present the CPU request via the Downward API
+			// k8s automatically rounds it up to the nearest integer unit.  That's how
+			// we plan to set this variable, but just in case it might be worth doing this
+			// check in case they ever break that.  There's no official contract for that.
+			millicoreRegexp := regexp.MustCompile(`^(P<millis>[1-9][0-9]*)m$`)
```

Comment:
`MustCompile` should only be used in global variable definition, and this regexp should be moved to be a global variable definition (as compiling is expensive and we only need to do it once)